ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/SqlExecutor.scala

```diff
@@ -28,11 +28,18 @@ class SqlExecutor(noOfThread: Int) extends AutoCloseable {
         .build()
     )
 
-  def runQuery[A](block: () => A): Future[A] = {
+  def runQuery[A](description: String, block: () => A): Future[A] = {
     val promise = Promise[A]
+    val startWait = System.nanoTime()
     executor.execute(() => {
       try {
-        promise.success(block())
+        val elapsedWait = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startWait)
+        val start = System.nanoTime()
+        val res = block()
+        val elapsed = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start)
+        logger.trace(
```

Comment:
If you decide to make `description` _call-by-name_ it would be better to use check that the level is enabled before accessing `description`. This would come for free if we're using [scala-logging](https://github.com/lightbend/scala-logging), do we use it?

Also the rest of the time tracking operations are relevant only if trace level logging is enabled. Rather then dotting this method with `if`s it would probably be better instead to dispatch to either of two underlying implementations based on the log level, one with time tracking and the other without.

Reply:
Given the suggestion at the second paragraph of the previous comment, it would probably be better to not use scala-logging but Logback directly to avoid macros injecting unnecessary conditionals in our code.

Reply:
I'll use call-by-name. I'm not worried about the overhead of the timing aspect. After all, we're talking about a network hop to the DB (with potentially high latency).

Reply:
Don't forget that `logger` can write to the most surprising places. :wink:

Reply:
Sorry, misread your comments. I wrapped the logging in `isTraceEnabled` in 7ae9da263.